### PR TITLE
Feat/update unit

### PR DESCRIPTION
**Pull request by maykremer**

### CODE REVIEW

File: mockserver/unit/unit.manager.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ class UnitManager {
   static getUnitById(req, res) {
     res.send(units.find((unit) => unit.id === req.params.id));
   }
+  static updateUnitById(req, res) {
+     const selectedUnit = units.find((unit) => unit.id === req.params.id);
```
Comment (carmeltsadok) [inline mockserver/unit/unit.manager.js:19]: update the object in the units array as well

**Review by PCOffline — APPROVED** [review]

File: src/pages/UserManagement/components/EditUnit/EditUnit.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { useTranslation } from 'react-i18next';
+import { Input, Button } from '@material-ui/core';
+import { useState } from 'react';
+import { toast } from 'react-toastify';
+import useStyles from './EditUnit.styles.js';
+import UnitService from '../../../../services/unit.service.js';
+import CustomDialog from '../../../../common/CustomDialog/CustomDialog.jsx';
+
+const EditUnit = ({ openEditUnit, setOpenEditUnit, unit, units, setUnits, setSelectedUnit }) => {
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/EditUnit/EditUnit.jsx:9]: edit unit and create unit dialog's content are exactly the same. 
create a component the switch between creation and update depends on its props, 
so there will be no repetition between the components (style and overall code)

File: src/pages/UserManagement/components/UnitDetailsHeadLine/UnitDetailsHeadLine.styles.js [hunk 3]
```diff
@@ -29,9 +29,14 @@ const useStyles = makeStyles((theme) => ({
...
+  unitName: {
+    color: 'white',
+    fontSize: '1.1rem',
+  },
   mainDivText: {
-    marginTop: '1.5rem',
-    marginLeft: '1rem',
+    display: 'flex',
+    alignItems: 'center',
+
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/UnitDetailsHeadLine/UnitDetailsHeadLine.styles.js:39]:
```suggestion
```

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/pages/UserManagement/components/EditUnit/EditUnit.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    UnitService.updateUnitById(unit.id, { name: newUnitName }).then((updatedUnit) => {
+        const unitIndex = units.findIndex(specUnitIndex);
+        const filteredUnits = units.filter((existUnit) => existUnit.id !== unit.id);
+        // insert the new unit to the same place
+        filteredUnits.splice(unitIndex, 0, updatedUnit);
+        setSelectedUnit(updatedUnit);
+        setUnits(() => [...filteredUnits]);
+        toast(t('text.unitEdited'));
+    }).catch(() => {
+        toast(t('text.uniNotEditWarning'));
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/EditUnit/EditUnit.jsx:21]: change to `unitNotEditWarning`

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Review by vitaly-furman — APPROVED** [review]

**Review by carmeltsadok — APPROVED** [review]

**Review by PCOffline — APPROVED** [review]

**Pull request merged**